targets/ublk.nfs.cpp

```diff
@@ -0,0 +1,383 @@
+/* -*-  mode:c; tab-width:8; c-basic-offset:8; indent-tabs-mode:nil;  -*- */
+// SPDX-License-Identifier: MIT or GPL-2.0-only
+
+#include <config.h>
+
+#include "ublksrv_tgt.h"
+#include <nfsc/libnfs.h>
+
+
+struct nfs_tgt_data {
+        char url[4096];
+	struct nfs_context *nfs;
+        struct nfsfh *nfsfh;
+        size_t capacity;
+};
+
+typedef struct nfs_cb_data {
+        struct nfs_cb_data *next;
+        const struct ublksrv_queue *q;
+        int tag;
+        ssize_t count;
+} nfs_cb_data_t;
+
+nfs_cb_data_t *io_list;
+
+pthread_spinlock_t io_list_lock;
+
```

Comment:
The above two global variables can be defined as per-queue data by implementing ->init_queue(), please refer to ublk.nbd.

Reply:
then ublk.nfs MQ could be possible, otherwise you may have to fail to setup in case of MQ from command line.

Reply:
Done. Thanks.